src/pudl/constants.py

```diff
@@ -1423,8 +561,14 @@
     into PUDL.
 """
 
-pudl_tables = {
-    'eia860': eia860_pudl_tables,
+PUDL_TABLES: Dict[str, Tuple[str]] = {
```

Comment:
Is this distinction necessary? Shouldn't the user be able to ask for any of the tables in the database? The docstring just says "Tables that are available in the PUDL DB", which is precisely what `RESOURCE_METADATA` is supposed to be. You say it should not contain tables that are derived / harvested, but it contains "glue" tables and entity tables. :/

Is it expected that `PUDL_TABLES` not contain `natural_gas_transport_eia923`, `accumulated_depreciation_ferc1`, `utilities_ferc1`, or `plants_ferc1` ?

Reply:
Right now this selection of tables from these lists is used as one of the main arguments determining what data is processed. The docstring is wrong. These tables are special because they are the primary source tables from which all the other data is derived, and each of them has a transform function dedicated to it, including the glue tables (though they are processed all together, as a function of which other datasets are being loaded and need to be glued together). Honestly I'm not sure what the entity tables are doing in here at the moment.

If you put `natural_gas_transport_eia923` or `utilities_ferc1` in there the system would have no idea what to do with it today. There's no function that generates that table in the extract/transform steps. I'm sure logic could be added that could infer which of these tables need to be processed based on what subset of the universe of output tables the user has asked for, but that sounds complicated.

In practice, unless we're doing development, we almost always process all of the tables within a dataset together at the same time, and really we don't want to expose all possible combinations of tables to the user, since we don't test them all and don't know if they'll actually work, and certainly tooling built on top of the DB will expect all of them to be present. So I think from a user point of view, what we want to enable is selecting a list of datasets (e.g. just the `eia176`, `ferc2`, and `phmsagas`) and a subset of the available partitions of each of those datasets (e.g. a set of years). But also we're moving away from expecting individual users who aren't involved in development to run the ETL at all, and instead trying to provide the outputs in a ready-to-use form.

On the other hand, in development it's very useful to be able to run a particular branch of the pipeline up to a certain point on a very restricted subset of the data, so you can debug it, tweak it, and re-run it, without having to wait for the whole thing including a bunch of data that is irrelevant to the current task.

So, the role that this little data structure plays is enumerating what subsets of the data can be processed independently. It's similar to the data partitions (years/states), except that it's slicing the data along semantic dimensions, not spacetime dimensions. I can imagine other ways to do this (and would be interested to hear what you think would be good ways to do it) but I think changing the way we do it would require an overhaul of the way the ETL pipeline works, and I think we might want to put that off a bit.